front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -246,13 +245,57 @@ const castNodeToNge = (
   ),
 });
 
-export const getTrainrunCategories = (t: TFunction<'operational-studies'>): TrainrunCategory[] =>
-  Array.from(OSRD_TRAINRUN_CATEGORY_MAPPING.entries()).map(([key, category]) => ({
+export const getTrainrunCategories = async (
+  t: TFunction<'operational-studies'>,
+  dispatch: AppDispatch
+): Promise<TrainrunCategory[]> => {
+  const subCategoryPage = await dispatch(
+    osrdEditoastApi.endpoints.getSubCategory.initiate({ page: 1 }, { subscribe: false })
+  ).unwrap();
+
+  const subCategories: SubCategory[] = [...subCategoryPage.results];
+
+  const formattedSubCategories = subCategories.map((sub) => ({
+    trainCategory: sub.name,
+    colorRef: sub.code,
+  }));
+
+  const OSRD_TRAINRUN_SUB_CATEGORY_CODE_MAPPING = new Map<SubCategory['code'], TrainrunCategory>();
+
+  formattedSubCategories.forEach(({ trainCategory, colorRef }, index) => {
+    OSRD_TRAINRUN_SUB_CATEGORY_CODE_MAPPING.set(trainCategory, {
+      id: OSRD_TRAINRUN_MAIN_CATEGORY_CODE_MAPPING.size + index + 1,
+      order: index + 1,
```

Comment:
```suggestion
      order: OSRD_TRAINRUN_MAIN_CATEGORY_CODE_MAPPING.size + index + 1,
```

Otherwise, different `TrainrunCategory` have the same `order` value. However, it looks like having same `order` values don't break anything in NGE, but I'd rather keep it clean.

For example, take a look at the `metadata.trainrunCategories.id` and `metadata.trainrunCategories.order` for the 2 last categories:
[reticulaire(14).json](https://github.com/user-attachments/files/21852405/reticulaire.14.json)

Reply:
Done